app/configurator/configurator-state.tsx

```diff
@@ -888,58 +913,54 @@ const reducer: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
 
         if (metadata) {
           const { dimensions, measures } = metadata;
-          const previousConfig = current(draft.chartConfig);
-          draft.chartConfig = getChartConfigAdjustedToChartType({
-            chartConfig: previousConfig,
+          const previousConfig = getChartConfig(draft);
+          const newConfig = getChartConfigAdjustedToChartType({
+            chartConfig: current(previousConfig),
             newChartType: chartType,
             dimensions,
             measures,
           });
-          draft.activeField = undefined;
-          draft.chartConfig = deriveFiltersFromFields(
-            draft.chartConfig,
-            metadata.dimensions
-          );
+
+          for (const k in previousConfig) {
```

Comment:
@ptbrowne it kinda worked with this, when I used the` draft.chartsConfigs[index] = newChartConfig`, there were some Proxy issues. I think it should be the same as when doing splice 🤔